mev_inspect/compound_liquidations.py

```diff
@@ -0,0 +1,113 @@
+    comptroller_instance = w3.eth.contract(
+        address=V2_COMPTROLLER_ADDRESS, abi=comp_v2_comptroller_abi
+    )
+    markets = comptroller_instance.functions.getAllMarkets().call()
+    for c_token in markets:
+        # make an exception for cETH (as it has no .underlying())
+        if c_token != V2_C_ETHER:
+            comp_v2_ctoken_abi = get_raw_abi("CToken", Protocol.compound_v2)
+            ctoken_instance = w3.eth.contract(address=c_token, abi=comp_v2_ctoken_abi)
+            underlying_token = ctoken_instance.functions.underlying().call()
+            c_token_mapping[
+                c_token.lower()
+            ] = underlying_token.lower()  # make k:v lowercase for consistancy
+    return c_token_mapping
+
+
+# # find if the searcher repays the loan from their own EOA, by buying it from a DEX, or w/ a flashloan
+# def find_collateral_source(
+#     classified_traces: list[ClassifiedTrace],
+#     tx: Transaction,
+#     liquidation_contract: Optional[str],
+# ) -> LiquidationCollateralSource:
+#     source = LiquidationCollateralSource.other  # set other by default
+#     for classified_trace in classified_traces:
+#         # look for trace that liquidates and see from address
+#         if (
+#             classified_trace.to_address == liquidation_contract
+#             and classified_trace.function_name == "liquidateBorrow"
+#         ):
+#             ## check if tx originates from searcher eoa or contract
+#             if tx.to_address.lower() == classified_trace.from_address:
+#                 source = LiquidationCollateralSource.searcher_contract
+#             elif tx.from_address.lower() == classified_trace.from_address:
+#                 source = LiquidationCollateralSource.searcher_eoa
+#             ## flashloan providers identified here
+#     return source
+
+
+def get_underlying_ctoken_exchange_rate(
+    c_token_address: str, block_number: int
+) -> float:
+    comp_v2_ctoken_abi = get_raw_abi("CToken", Protocol.compound_v2)
+    ctoken_instance = w3.eth.contract(address=c_token_address, abi=comp_v2_ctoken_abi)
+    raw_exchange_rate = ctoken_instance.functions.exchangeRateCurrent().call(
+        block_identifier=block_number
+    )
+    # format based on decimals in ctoken and the underlying token
+    # see "Interpreting Exchange Rates" https://compound.finance/docs#protocol-math
+    underlying_token_address = get_all_comp_markets()[c_token_address.lower()]
+    decimals_in_underlying = get_erc20_token_decimals(
+        Web3.toChecksumAddress(underlying_token_address)
+    )
+    decimals_in_ctoken = get_erc20_token_decimals(c_token_address)
+    return raw_exchange_rate / (
+        10 ** (18 + decimals_in_underlying - decimals_in_ctoken)
+    )
+
+
+# CEther liquidations for now, CToken liquidations have a slightly different format
+def get_cether_liquidations(
+    classified_traces: list[ClassifiedTrace],
+) -> List[Liquidation]:
+    comp_markets = get_all_comp_markets()
```

Comment:
can this function take in `comp_markets` instead?

that would let this be fully static instead of making an internal network call which makes scaling easier